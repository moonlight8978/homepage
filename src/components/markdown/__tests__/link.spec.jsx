import React from 'react'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import Link from '../link'

test('render correctly with external URL', () => {
  const { container } = render(
    <Link href="https://google.com">This is a link</Link>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('render correctly with local path', () => {
  const { container } = render(
    <MemoryRouter>
      <Link href="/local-path">This is a link</Link>
    </MemoryRouter>
  )

  expect(container.firstChild).toMatchSnapshot()
})
