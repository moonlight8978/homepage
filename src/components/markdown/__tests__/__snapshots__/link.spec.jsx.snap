// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with external URL 1`] = `
<a
  class="link"
  data-testid="markdown-link"
  href="https://google.com"
>
  This is a link
</a>
`;

exports[`render correctly with local path 1`] = `
<a
  class="link"
  data-testid="markdown-link"
  href="/local-path"
>
  This is a link
</a>
`;
